library/psa_crypto.c

```diff
@@ -5219,6 +5227,32 @@ static psa_status_t psa_tls12_prf_set_key( psa_tls12_prf_key_derivation_t *prf,
     return( PSA_SUCCESS );
 }
 
+static psa_status_t psa_tls12_prf_set_other_key( psa_tls12_prf_key_derivation_t *prf,
```

Comment:
We are in the block that's common to `PSA_ALG_TLS12_PRF` and `PSA_ALG_TLS12_PSK_TO_MS`, but I don't think that's correct, since `other_secret` is specific to the second.

I think this should be renamed to `psa_tls12_prf_psk_to_ms_set_other_key()` and moved to the block that's specific to `PSA_ALG_TLS12_PSK_TO_MS` - together with `psa_tls12_prf_psk_to_ms_set_key()` and `psa_tls12_prf_psk_to_ms_input()`.

Also, the dispatch of the `other_secret` state should be done in `psa_tls12_prf_psk_to_ms_input()`, not `psa_tls12_prf_input()` which should keep rejecting `INPUT_OTHER_SECRET`.

Just to make sure, I'd like to have a negative test case ensuring that an attempt to pass an `INPUT_OTHER_SECRET` while using `PSA_ALG_TLS12_PRF` (as opposed to `PSA_ALG_TLS12_PSK_TO_MS`) indeed fails as expected.